src/common/Smi.Common/Options/SmiCliInit.cs

```diff
@@ -120,6 +117,16 @@ public static T Verify<T>(object parsedOptions)
             return asExpected;
         }
 
+        private static string GetHostProcessName(Type t)
+        {
+            string hostProcessName = t.Namespace?.Split('.')[1];
```

Comment:
Not sure but I think that `?.` will return null which you then index using [1].  So either drop the `?` or use `t.Namespace?.Split('.')?.Skip(1)?.FirstOrDefault()`

Reply:
This should work fine for all our existing apps (have tested manually), but the issue was the tests don't follow the same execution path which sets the HostProcessName elsewhere. Have fixed this now I think.

Reply:
Oh that's cool, didn't know that worked but seems it does

```
string s = null ;
string b = s?.Split('.')[1];
```
_Gives b==null_